cloud/google/machineactuator.go

```diff
@@ -439,11 +439,18 @@ func (gce *GCEClient) Update(cluster *clusterv1.Cluster, goalMachine *clusterv1.
 		}
 	}
 
+	currentConfig, err := gce.machineproviderconfig(currentMachine.Spec.ProviderConfig)
+	if err != nil {
+		return gce.handleMachineError(currentMachine, apierrors.InvalidMachineConfiguration(
+			"Cannot unmarshal machine's providerConfig field: %v", err), createEventAction)
```

Comment:
We are in the Update function so createEventAction seems wrong here (note that on line 417 noEventAction is used.

Reply:
fixed.